user-guide/modules/ROOT/pages/advanced-builds.adoc

```diff
@@ -1,3 +1,247 @@
 = Advanced Builds
+:navtitle: Advanced Builds
+
+This section covers a number of advanced build scenarios that you may come across if your project has complex, legacy, or non-standard, components or processes.
+
+== Build with a Non-default Compiler
+
+Using the `b2` build system, you can specify the compiler you want to use with the `toolset` option. For example, if you want to use the GNU Compiler Collection (GCC), you would specify `toolset=gcc`. If you want to use Clang, you would use `toolset=clang`. For example, to build the Boost libraries with gcc:
+
+[source,txt]
+----
+./b2 toolset=gcc
+----
+
+If you want to use a specific _version_ of a compiler, you can specify it after the compiler name, separated by a dash. For example, to use GCC 7.3, you would use `toolset=gcc-7.3`:
+
+[source,txt]
+----
+./b2 toolset=gcc-7.3
+----
+
+You can also specify the toolset option when you're building your application. This is done in a similar way, by passing `toolset=` to the `b2` command, as part of your build process. Here's an example:
+
+[source,txt]
+----
+b2 toolset=clang
+----
+
+If the compiler is not in your system's PATH, you will need to provide the full path to the compiler executable. If there are spaces in the path, then wrap it in quotes, for example:
+
+[source,txt]
+----
+b2 toolset="C:\Program Files (x86)\Clang\"
+----
+
+== Specify the Application Binary Interface
+
+Building Boost with the old Application Binary Interface (ABI)) refers to compiling the library with a flag to ensure backward compatibility with an older version of an ABI.
```

Comment:
This subsection could move to the end. It's way more specific than the previous and next subsections. It's also not particularly related to boost because this is only a flag and how to set flags in general is the only thing related to boost b2 here. Setting the ABI is also way more specific than the use cases of other examples.

Reply:
ok